beta-pipelines/skylab/m3c/CondensedSnm3C.wdl

```diff
@@ -468,27 +473,78 @@ task Split_unmapped_reads {
     }
 }
 
-#task hisat_single_end_r1_r2_mapping_dna_mode_and_merge_sort_split_reads_by_name {
-#    input {
-#        File split_r1
-#        File split_r2
-#    }
-#    command <<<
-#    >>>
-#    runtime {
-#        docker: "fill_in"
-#        disks: "local-disk ${disk_size} HDD"
-#        cpu: 1
-#        memory: "${mem_size} GiB"
-#    }
-#    output {
-#        File r1_hisat3n_bam = ""
-#        File r1_hisat3n_stats = ""
-#        File r2_hisat3n_bam = ""
-#        File r2_hisat3n_stats = ""
-#        File merge_sorted_bam = ""
-#    }
-#}
+task Hisat_single_end_r1_r2_mapping_dna_mode_and_merge_sort_split_reads_by_name {
+    input {
+        File split_fq_tar
+        File genome_fa
+        File tarred_index_files
+
+        String docker = "us.gcr.io/broad-gotc-prod/m3c-yap-hisat:1.0.0-2.2.1"
+        Int disk_size = 80
+        Int mem_size = 20
+    }
+    command <<<
+        set -euo pipefail
+
+        mkdir reference/
+
+        cp ~{tarred_index_files} reference/
+        cp ~{genome_fa} reference/
+
+        # untar the tarred index files
+        cd reference/
+        tar -xvf ~{tarred_index_files}
+        rm ~{tarred_index_files}
+        samtools faidx hg38.fa
+
+        # untar the unmapped fastq files
+        tar -xvf ~{split_fq_tar}
+        rm ~{split_fq_tar}
+
+        # define lists of r1 and r2 fq files
+        R1_files=($(ls | grep "\.hisat3n_dna.split_reads.R1.fastq"))
+        R2_files=($(ls | grep "\.hisat3n_dna.split_reads.R2.fastq"))
+
+        for file in "${R1_files[@]}"; do
+          sample_id=$(basename "$file" ".hisat3n_dna.split_reads.R1.fastq")
+          hisat-3n /cromwell_root/reference/hg38 -q -U ${sample_id}.hisat3n_dna.split_reads.R1.fastq --directional-mapping-reverse --base-change C,T --no-repeat-index --no-spliced-alignment --no-temp-splicesite -t --new-summary --summary-file ${sample_id}.hisat3n_dna_split_reads_summary.R1.txt --threads 11 | samtools view -b -q 10 -o "${sample_id}.hisat3n_dna.split_reads.R1.bam"
+        done
+
+       for file in "${R2_files[@]}"; do
+         sample_id=$(basename "$file" ".hisat3n_dna.split_reads.R2.fastq")
+         hisat-3n /cromwell_root/reference/hg38 -q -U ${sample_id}.hisat3n_dna.split_reads.R2.fastq --directional-mapping --base-change C,T --no-repeat-index --no-spliced-alignment --no-temp-splicesite -t --new-summary --summary-file ${sample_id}.hisat3n_dna_split_reads_summary.R2.txt --threads 11 | samtools view -b -q 10 -o "${sample_id}.hisat3n_dna.split_reads.R2.bam"
```

Comment:
Same here (split the command over multiple lines for readability and consistency)